src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -193,35 +260,396 @@ IMF_STD_ATTRIBUTE_DEF (altitude, Altitude, float)
 //
 
 IMF_STD_ATTRIBUTE_DEF (expTime, ExpTime, float)
 
 
+//
+// focalLength -- the focal length of the lens, in millimeters, when
+// the image was created or captured
+//
+// This represents nominal focal length, i.e. the number printed on
+// the barrel of a prime (i.e. fixed-focal-length) lens, or the number
+// indicated by the position of the focal length ring on a zoom lens.
+// The latter is typically approximate and zoom lenses may emit lens
+// metadata giving a more precise nominal focal length.
+// 
+
+IMF_STD_ATTRIBUTE_DEF (focalLength, FocalLength, float)
+
+
+//
+// effectiveFocalLength -- focal length in millimeters of a hypothetical
+// lens the field of view of which matches the actual field of view of the
+// lens used in creating or capturing the image and which exhibits
+// no 'focus breathing'
+//
+
+IMF_STD_ATTRIBUTE_DEF (effectiveFocalLength, EffectiveFocalLength, float)
+
+
 //
 // aperture -- the camera's lens aperture, in f-stops (focal length
 // of the lens divided by the diameter of the iris opening)
+// 
+// SMPTE ST 2065-4:2013 defines ACES Container file where the attribute
+// named 'aperture' can contain either the T-stop of the lens, or the
+// F-number of the lens.
+//
+// If the T-stop is known, it is preferable to use the tStop attribute
+// to indicate this unambiguously.
+//
+// If the f-Number is known, it is preferable to use the fNumber
+// attribute to indicate this unambiguously to OpenEXR readers following
+// strict ACES Container semantics.
+// 
+// If a number is provided to an application creating an OpenEXR file
+// but it's unclear whether that number is a T-stop or an F-number, the
+// aperture attribute should be used to carry that metadatum.
 //
+// If T-stop and/or f-Number is/are knwown, the aperture attribute
+// should be omitted unless existing downstream processes depend on its
+// presence instead of the more precisely-defined attribute(s).
 
 IMF_STD_ATTRIBUTE_DEF (aperture, Aperture, float)
 
 
+//
+// fNumber -- ratio of lens focal length to diameter of lens entrance
+// pupil at the time the image was created or captured
+// 
+// The lens focal length used in the calculation is the nominal focal
+// length which would be stored in the focalLength attribute, even if
+// the effective focal length (compensating for any focus breathing)
+// is known and stored in an effectiveFocalLength attribute.
+//
+
+IMF_STD_ATTRIBUTE_DEF (fNumber, FNumber, float)
+
+
+//
+// tStop -- ratio of lens focal length to diameter, adjusted for
+// transmittance
+
+IMF_STD_ATTRIBUTE_DEF (tStop, TStop, float)
+
+
 //
 // isoSpeed -- the ISO speed of the film or image sensor
 // that was used to record the image
 //
+// Techniques to determine the ISO speed rating of a digital still camera
+// are given in ISO 12232:2019. In the absence of any comparable
+// specification for a digital motion picture camera the same procedures
+// can be used. For film, the manufacturer's stated ISO is appropriate.
+//
```

Comment:
ISO 12232:2019 is hidden behind a paywall, and basically has techniques to gin up a value that corresponds in some way to film's ISO, and certainly DSLRs and other cameras will provide an ISO value that makes sense in this context. It seems that to fully capture useful attributes, one would also want to record gain values for video cameras? I realize that's likely outside the mien of 2065-4:2013, but...

Reply:
Thank you for making me acknowledge my privilege (I mean that sincerely); ISO standards are out of reach of many people who might read this specification. On reflection it would be better to avoid mentioning anything other that the name of that organization. So:

//
// isoSpeed -- measure of sensitivity of the camera system to light, as established by the ISO (International Standards Organization)
//

// For a film-based camera system, the ISO speed will be marked on the film packaging by the film manufacturer;
// for a digital camera system, the ISO speed will typically be a camera setting indicating some combination of
// electronic and/or digital signal processing altering the relationship between light impinging on the sensor at
// a given point, and the digital value recording that impingement.
//
// A camera system's ISO speed is also sometimes described as its 'exposure index'.

How does that sound?

For comparison the definition in the standard is below.

![image](https://user-images.githubusercontent.com/1134845/103465271-f70f0a80-4d07-11eb-8bab-78690cf7581e.png)

Reply:
Your proposed version works very well for me, thanks!